tests/lax_numpy_test.py

```diff
@@ -5379,6 +5379,53 @@ def testIssue2347(self):
     np_object_list = np.array(object_list)
     self.assertRaises(TypeError, jnp.array, np_object_list)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}".format(
+          jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype}
+      for shape in all_shapes
+      for dtype in complex_dtypes))
+  def testLogaddexpComplex(self, shape, dtype):
+    rng = jtu.rand_some_nan(self.rng())
+    args_maker = lambda: (rng(shape, dtype), rng(shape, dtype))
+
+    @jtu.ignore_warning(category=RuntimeWarning, message="invalid value.*")
+    def np_fun(x1, x2):
+      return np.log(np.exp(x1) + np.exp(x2))
+
+    @jtu.ignore_warning(category=RuntimeWarning, message="invalid value.*")
+    def jnp_fun(x1, x2):
+      y = jnp.logaddexp(x1, x2)
+      y = y.real + ((y.imag + np.pi) % (2 * np.pi) - np.pi) * 1j
```

Comment:
As written this would allow a future refactoring of `jnp.logaddexp` to return different outputs and still have the test pass. I think the logic for standardizing the complex branch of the output should go within the `logaddexp` function rather than its test. If possible, we should anticipate what branch `np.logaddexp` will likely use when it is implemented.

One idea: could we somehow ensure that `logaddexp(x1, x2)` will always choose the same branch as `log(x1 + x2)`?

Reply:
My main concern is that adding a modulo operation into `jnp.logaddexp` will cause a little performance loss. Besides that, scipy's complex `logsumexp` doesn't seem to have such standardization, although it claims that it behaves the same as `np.log(np.sum(np.exp(a)))` in the document.

Actually, if we really want to fully reproduce the behavior of `np.log`, it requires more than a modulo, because the imaginary part of `np.log` can range in `[-pi, +pi]`, with both ends of the interval closed. For example, `np.log(np.pi*1j) = +pi`, and `np.log(-np.pi*1j) = -pi`.

Reply:
Sure, that makes sense if you want to keep the implementation as-is. But lets test the output in a way that validates the output of the function, rather than validating a post-processed version of the output.

Reply:
What are you suggesting we do for the test, then? 
It's rather hard to predict, given two outputs, what branch the output of `jnp.logaddexp`.

The sensible test should be to check that real parts match and imagnary parts differ by a multiple of 2*pi.
Should we avoid using `_CheckAgainstNumpy` and implement something like `_CheckAgainstNumpy2PiImagPhase`?

Reply:
If you agree that we will not standardize the branch of `jnp.logaddexp`, the valid test for complex log should not be 'real and image parts are the same', but 'real parts are the same, and image parts mod 2 pi are the same'. So I still think we need to first post-process the output by taking the mod, then send it to `_CheckAgainstNumpy`.

Maybe we should not use `_CheckAgainstNumpy` here, because its semantic would be 'real and image parts are the same'... but if we write our own check here, we would just copy a lot of code from it.

Reply:
> The branch may also change if we refactor the implementation in future.

This is what I'm hoping the test would catch. Over time users tend to write code that relies on such implementation details; it would be nice if the tests would let us know of such changes if and when they happen.

Reply:
Ok, then I will put the mod into `jnp.logaddexp`.